### PR TITLE
BMR16 Login screen

### PR DESCRIPTION
**Pull request by kamdaou-amalitech**
Created login screen, the ViewModel, and tests.
Test written
Email and password validation

### CODE REVIEW

File: ui/authentication/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+apply {
+    from("$rootDir/compose-module.gradle")
+}
+
+dependencies {
+    "implementation"(project(Modules.core))
+    "implementation"(project(Modules.coreUI))
+    "implementation"(project(Modules.domainAuthentication))
+
+    "implementation"(Lifecycle.composeLifecycle)
```
Comment (bensek) [inline ui/authentication/build.gradle.kts:10]: This doesn't look fixed. Removed the lifecycle from here.

Comment (kamdaou-amalitech) [reply ui/authentication/build.gradle.kts:10]: I use that dependency in LoginScreen, line 62. If I remove it, I will not be able to access collectAsStateWithLifecycle.

Comment (bensek) [reply ui/authentication/build.gradle.kts:10]: Add it to the base compose file other features need it too.

**Review by bensek — APPROVED** [review]

File: app/build.gradle.kts [hunk 2]
```diff
@@ -96,6 +96,8 @@ dependencies {
     implementation(project(Modules.domainOnboarding))
     implementation(project(Modules.uiOnboarding))
     implementation(project(Modules.coreUI))
+    implementation(project(Modules.uiAuthentication))
```
Comment (bensek) [inline app/build.gradle.kts:99]: Cadet, Login, SignUp,Forgot Password are all supposed to be part of the onboarding module. It shouldnt be a separate module. It's all the same feature.

**Review by bensek — CHANGES_REQUESTED** [review]

**Review by esthcarelle — APPROVED** [review]

**Review by bensek — APPROVED** [review]

**Pull request merged**